terraform/cloudfront/cloudfront_distribution/distribution.tf

```diff
@@ -0,0 +1,62 @@
+resource "aws_route53_record" "cdn_domain" {
+  count = "${length(var.aliases)}"
+
+  zone_id = "${var.system_dns_zone_id}"
+  name    = "${element(var.aliases, count.index)}."
+  type    = "CNAME"
+  ttl     = "60"
+  records = ["${aws_cloudfront_distribution.cdn_instance.domain_name}"]
+}
+
+resource "aws_cloudfront_distribution" "cdn_instance" {
+  origin {
+    domain_name = "${var.origin}"
+    origin_id   = "${var.origin}"
+
+    custom_origin_config {
+      origin_protocol_policy = "https-only"
+      http_port              = 80
+      https_port             = 443
+      origin_ssl_protocols   = ["TLSv1", "TLSv1.1", "TLSv1.2"]
+    }
+  }
+
+  aliases         = "${var.aliases}"
+  comment         = "${var.comment}"
+  enabled         = true
+  is_ipv6_enabled = true
+
+  default_cache_behavior {
+    allowed_methods  = ["DELETE", "GET", "HEAD", "OPTIONS", "PATCH", "POST", "PUT"]
+    cached_methods   = ["GET", "HEAD"]
+    target_origin_id = "${var.origin}"
+
+    forwarded_values {
+      query_string = true
+
+      cookies {
+        forward = "none"
+      }
+    }
+
+    viewer_protocol_policy = "redirect-to-https"
+    min_ttl                = 0
+    default_ttl            = 3600
+    max_ttl                = 86400
+  }
+
+  restrictions {
+    geo_restriction {
+      restriction_type = "none"
+    }
+  }
+
+  tags {
+    Environment = "${var.env}"
+  }
+
+  viewer_certificate {
+    acm_certificate_arn            = "${var.system_domain_cert_arn}"
+    cloudfront_default_certificate = false
+  }
```

Comment:
I think this is missing a `ssl_support_method` param, which is required when specifying a cert ([see docs](https://www.terraform.io/docs/providers/aws/r/cloudfront_distribution.html#ssl_support_method)). This need to be set to one of `vip` or `sni-only`. I think we can probably use `sni-only`.

At present, I get the following error:
```
2 error(s) occurred:

* aws_cloudfront_distribution.cdn_instance: MalformedXML: 1 validation error detected: Value '' at 'distributionConfigWithTags.distributionConfig.viewerCertificate.sSLSupportMethod' failed to satisfy constraint: Member must satisfy enum value set: [sni-only, vip]
        status code: 400, request id: be2f4fc6-ea00-11e6-b6c6-29d0a4c8b7f3
* aws_cloudfront_distribution.cdn_instance: MalformedXML: 1 validation error detected: Value '' at 'distributionConfigWithTags.distributionConfig.viewerCertificate.sSLSupportMethod' failed to satisfy constraint: Member must satisfy enum value set: [sni-only, vip]
        status code: 400, request id: be2f9e4f-ea00-11e6-b089-c378731fc466
```